contracts/modules/social_recovery/storage/GuardianStorage.sol

```diff
@@ -53,21 +58,30 @@ contract GuardianStorage is IGuardianStorage {
             entry.guardians[SENTINEL_GUARDIANS] = _guardian;
         }
         entry.count++;
+        emit GuardianAdded(_wallet, _guardian);
+        if (entry.threshold != _threshold){
+            _changeThreshold(_wallet, _threshold);
+        }
     }
 
     /**
-     * @dev Lets an authorised module revoke a guardian from a wallet.
+     * @dev Lets an authorised module revoke a guardian from a wallet and change the threshold.
      * @param _wallet The target wallet.
      * @param _prevGuardian Guardian that pointed to the guardian to be removed in the linked list
      * @param _guardian The guardian to revoke.
      */
-    function revokeGuardian(address _wallet, address _prevGuardian, address _guardian) external onlyModule(_wallet) {
+    function revokeGuardianWithThreshold(address _wallet, address _prevGuardian, address _guardian, uint256 _threshold) external onlyModule(_wallet) {
         GuardianStorageEntry storage entry = entries[_wallet];
         require(_guardian != address(0) && _guardian != SENTINEL_GUARDIANS, "GS: invalid guardian");
         require(entry.guardians[_prevGuardian] == _guardian, "GS: invalid previous guardian");
+        require(entry.count - 1 >= _threshold, "GS: invalid threshold");
```

Comment:
```
require(entry.count > _threshold, "GS: invalid threshold");
```

Reply:
I believe this check is the same as the one already in the code, `entry.count >` and `entry.count - 1 >=`, can you clarify if there is something we're missing ?

Reply:
Yes, but the suggested one does two operations less (ADD and EQ) compared to the one mentioned in the code. AFAIU there are no security implications for this change as well.